### ISSUE TITLE
GPS not enabled on LineageOS 18.1

### ISSUE THREAD

**Issue by jech** [organicmaps/organicmaps#5999]
Moto G5 "cedric", Organic Maps 2023.18.8-FDroid, LineageOS 18.1 (no gapps).

When starting OM, the location indicator in the top bar does *not* appear.  The blue icon spins, until OM shows the dialog asking whether to continue.  A workaround is to first get a GPS fix in OSMAnd, then to immediately switch to OM — the GPS then works reliably.  The symptoms are identical to the ones described in #1853.

I have checked that the GPS works fine in OSMAnd, in Satstat, and in GPSTest.

**Comment by jech** [comment]
> Might this be a duplicate of this issue #4223 ?

Yep, it looks like it is.

**Comment by jech** [comment]
Looking at logcat, and I can see the following warnings:
```
09-15 23:25:40.433  2762  2780 W GnssLocationProvider: sendExtraCommand: unknown command force_xtra_injection
```

```
09-15 23:25:40.450   611   611 I LocSvc_GnssInterface: ===> start line 286 
09-15 23:25:40.454   611   657 E LocSvc_APIClientBase: onResponseCb:894] ERROR: 5 ID: 44
```

```
09-15 23:26:30.492  2762  2780 I GnssLocationProvider: GNSS HAL Requesting location updates from network provider for 10000 millis.
09-15 23:26:30.493  2762  2780 W GnssLocationProvider: Unable to request location.
09-15 23:26:30.493  2762  2780 W GnssLocationProvider: java.lang.IllegalArgumentException: provider doesn't exist: network
09-15 23:26:30.493  2762  2780 W GnssLocationProvider:  at com.android.server.location.LocationManagerService.requestLocationUpdatesLocked(LocationManagerService.java:1964)
09-15 23:26:30.493  2762  2780 W GnssLocationProvider:  at com.android.server.location.LocationManagerService.requestLocationUpdates(LocationManagerService.java:1947)
09-15 23:26:30.493  2762  2780 W GnssLocationProvider:  at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1178)
09-15 23:26:30.493  2762  2780 W GnssLocationProvider:  at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1139)
09-15 23:26:30.493  2762  2780 W GnssLocationProvider:  at com.android.server.location.gnss.GnssLocationProvider.handleRequestLocation(GnssLocationProvider.java:828)
09-15 23:26:30.493  2762  2780 W GnssLocationProvider:  at com.android.server.location.gnss.GnssLocationProvider.access$2700(GnssLocationProvider.java:100)
09-15 23:26:30.493  2762  2780 W GnssLocationProvider:  at com.android.server.location.gnss.GnssLocationProvider$ProviderHandler.handleMessage(GnssLocationProvider.java:2079)
09-15 23:26:30.493  2762  2780 W GnssLocationProvider:  at android.os.Handler.dispatchMessage(Handler.java:106)
09-15 23:26:30.493  2762  2780 W GnssLocationProvider:  at android.os.Looper.loop(Looper.java:223)
09-15 23:26:30.493  2762  2780 W GnssLocationProvider:  at android.os.HandlerThread.run(HandlerThread.java:67)
09-15 23:26:30.493  2762  2780 W GnssLocationProvider:  at com.android.server.ServiceThread.run(ServiceThread.java:44)
```

**Comment by biodranik** [comment]
Your logs show that the system reports the network location provider as available for the app, but in reality, the system does not have it.
1. Isn't it a bug of Lineage OS? Are there any related reports on their forums?
2. Did you configure the network provider somehow in the system settings? Is it enabled? If yes, can you try to disable it in the system?

**Comment by jech** [comment]
> Isn't it a bug of Lineage OS?

Quite possibly.  All I can report is that the GPS works (albeit slowly, since there's no AGPS) with OSMAnd, with GPSTest, and with SatStat.

> Did you configure the network provider somehow in the system settings?

No.  Unless I've missed something, there's no mention of a network provider anywhere in the settings.

Under "Location", there's an submenu "Wi-Fi and Bluetooth scanning", and both "Wi-Fi" and "Bluetooth" are set to "Off".  (As far as I understand, this setting does not enable a location provider, it merely allows applications to perform scans when Wi-Fi and Bluetooth are off.)

**Comment by jech** [comment]
> Your logs show that the system reports the network location provider as available for the app

Could you please point me at the exact place in the logcat that indicates that, and I'll file a bug with Lineage.

**Comment by biodranik** [comment]
@jech this short log says that OM tries to use the network provider. In the full OM log all providers available in the system are listed at the top. So I see two issues here:
1. System reports network provider as available (while it's not). That is a Lineage OS bug.
2. GPS doesn't work in OM (that is our issue, but it may be caused by the incorrect system behavior). We can try to build a debug or beta version _without_ a network provider at all for example and see how it works. @rtsisyk any other ideas why GPS is not working here?

**Comment by jech** [comment]
@biodranik, I've managed to install UnifiedNLP (which was quite a challenge, they really want you to install the full microg suite), and it looks like it has worked around the issue.  So my personal issue is apparently fixed.  Thanks for your help.

Still, it would be good to get to the bottom of the issue, and ensure that OM works on unmodified LineageOS installs.

**Comment by biodranik** [comment]
@rtsisyk can help here, if he is not too busy...

**Comment by jech** [comment]
After more testing, my installation of UnifiedNLP is too unstable to be useful.  So I'm unfortunately back with OSMAnd.

If someone modifies OM to fail gracefully when the network provider is not functional, they'll make my life somewhat more pleasant.

**Comment by biodranik** [comment]
We can detect if it's Lineage and never use the network provider, as a temporary workaround until the fix proposed by @rtsisyk is implemented: to launch both GPS and network providers at the same time.

**Comment by rtsisyk** [comment]
@Altonss @jech does Organic Maps work for you using pure GPS, without any any network locations enabled in the system?

**Comment by jech** [comment]
Without a network provider, Organic Maps fails to enable the GPS.  It runs fine if I enable the GPS by running GPStest in the background.

**Comment by rtsisyk** [comment]
> Without a network provider, Organic Maps fails to enable the GPS. It runs fine if I enable the GPS by running GPStest in the background.

Organic Maps without Google Service (your cases) tries ALL available providers in the system, including GPS. Please note that the cold start of GPS can take some time and Organic Maps turns off location services when you switch to another app. Please try to start navigation to keep GPS searching even when the app is on the screen.

**Comment by biodranik** [comment]
@Altonss @jech can you please share logs from the recent OM FDroid version?

**Comment by jech** [comment]
> Organic Maps without Google Service (your cases) tries ALL available providers in the system, including GPS. Please note that the cold start of GPS can take some time and Organic Maps turns off location services when you switch to another app.

Once again: the location icon does *not* appear in the status bar, indicating that the GPS is not being enabled.  I can get a GPS fix in 30s with GPSTest running in the background, but I never get a fix with OM alone.  My version of OM claims to be 2023.09.22-27-FDroid, which is the most recent one at the time of writing.

**Comment by jech** [comment]
> can you please share logs from the recent OM FDroid version?

Here's what I think is the relevant bit.  I can send you the full log by mail (I'm not willing to publish it, since it might contain private data).
```
10-29 22:46:19.415  8472  8472 I app.organicmaps.location: 
10-29 22:46:19.418   601   601 I MotoSensors: Sensorhub hal setDelay: 19 - 66
10-29 22:46:19.418   601   601 I MotoSensors: Sensorhub hal setdelay: 19 - 5
10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateAccelRate 200
10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateGyroRate 200
10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateMagRate 200
10-29 22:46:19.419   601   601 I MotoSensors: Sensorhub hal enable: 19 - 1
10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateAccelRate 10
10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateGyroRate 5
10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateMagRate 20
10-29 22:46:19.420  8472  8472 I app.organicmaps.location: provider = AndroidNativeProvider mInFirstRun = false oldInterval = 0 interval = 3000
10-29 22:46:19.426  8472  8472 I MwmApplication: Framework initialized
10-29 22:46:19.429   612   612 I LocSvc_GnssInterface: ===> setPositionMode_1_1 line 471 
10-29 22:46:19.429   612   660 E LocSvc_APIClientBase: onResponseCb:894] ERROR: 5 ID: 57
10-29 22:46:19.429   612   612 I LocSvc_GnssInterface: ===> start line 286 
10-29 22:46:19.429   612   612 I LocSvc_APIClientBase: locAPIStartTracking:342] start new session: 58
10-29 22:46:19.429   612   662 I LocSvc_api_v02: ---> locClientSendReq line 2325 QMI_LOC_REG_EVENTS_REQ_V02
10-29 22:46:19.435   612   662 I LocSvc_api_v02: ---> locClientSendReq line 2325 QMI_LOC_SET_OPERATION_MODE_REQ_V02
10-29 22:46:19.435   612  1599 I LocSvc_ApiV02: <--- globalRespCb line 213 QMI_LOC_SET_OPERATION_MODE_REQ_V02
10-29 22:46:19.436   612   662 I LocSvc_api_v02: ---> locClientSendReq line 2325 QMI_LOC_START_REQ_V02
10-29 22:46:19.436   601   816 D MotoSensors: gyroIntegration: quatMul bad
10-29 22:46:19.436  8472  8472 W onAccuracyChanged: Sensor android.sensor.rotation_vector has changed accuracy to 3
10-29 22:46:19.438   616   616 I android.hardware.power-service-qti: Power setMode: 5 to: 0
10-29 22:46:19.441   601   601 D MotoSensors: gyroIntegration: quatMul bad
10-29 22:46:19.442   612  1599 I LocSvc_ApiV02: <--- globalEventCb line 185 QMI_LOC_EVENT_WIFI_REQ_IND_V02
10-29 22:46:19.442   612  1599 I LocSvc_ApiV02: <--- globalEventCb line 185 QMI_LOC_EVENT_WIFI_REQ_IND_V02
10-29 22:46:19.442   612   660 I LocSvc_GnssInterface: ===> odcpiRequestCb line 513 
10-29 22:46:19.445   601   601 D MotoSensors: gyroIntegration: quatMul bad
10-29 22:46:19.447  2732  2749 I GnssLocationProvider: GNSS HAL Requesting location updates from network provider for 10000 millis.
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: Unable to request location.
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: java.lang.IllegalArgumentException: provider doesn't exist: network
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.location.LocationManagerService.requestLocationUpdatesLocked(LocationManagerService.java:1964)
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.location.LocationManagerService.requestLocationUpdates(LocationManagerService.java:1947)
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1178)
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1139)
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.location.gnss.GnssLocationProvider.handleRequestLocation(GnssLocationProvider.java:828)
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.location.gnss.GnssLocationProvider.access$2700(GnssLocationProvider.java:100)
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.location.gnss.GnssLocationProvider$ProviderHandler.handleMessage(GnssLocationProvider.java:2079)
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at android.os.Handler.dispatchMessage(Handler.java:106)
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at android.os.Looper.loop(Looper.java:223)
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at android.os.HandlerThread.run(HandlerThread.java:67)
10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.ServiceThread.run(ServiceThread.java:44)
10-29 22:46:19.450   601   601 D MotoSensors: gyroIntegration: quatMul bad
10-29 22:46:19.504   601   601 I chatty  : uid=1000(system) sensors@1.0-ser identical 11 lines
```

**Comment by biodranik** [comment]
@jech I think it was already discussed that the network location provider is broken in LineageOS. Can you somehow disable it completely in system settings and try again? A full log at fdroid@organicmaps.app would be also useful.

@rtsisyk the latest FDroid version should properly work with GPS despite a broken system network provider. For some reason, OSMand and other apps are working in this case.

**Comment by jech** [comment]
> I think it was already discussed that the network location provider is broken in LineageOS. Can you somehow disable it completely in system settings and try again?

This is the version of LineageOS, and there is no mention of the network provider anywhere in the settings, as mentioned in https://github.com/organicmaps/organicmaps/issues/5999#issuecomment-1733755334.  Installing a network provider does fix the issue in OM, but for various reasons it is not a useful workaround in my case (the older network providers are unstable under recent Android versions, while the newer ones require microg, which I don't want on my device).

**Comment by biodranik** [comment]
Thanks for the link. @rtsisyk why our code doesn't check if network provider is enabled before subscribing to it? The call to `LocationManager.isProviderEnabled(provider)` is missing:

```
  // A permission is checked externally
  @Override
  @RequiresPermission(anyOf = {ACCESS_COARSE_LOCATION, ACCESS_FINE_LOCATION})
  public void start(long interval)
  {
    Logger.d(TAG);
    if (!mProviders.isEmpty())
      throw new IllegalStateException("Already started");

    final LocationRequestCompat locationRequest = new LocationRequestCompat.Builder(interval)
        // The quality is a hint to providers on how they should weigh power vs accuracy tradeoffs.
        .setQuality(LocationRequestCompat.QUALITY_HIGH_ACCURACY)
        .build();

    // API 31+ provides `fused` provider which aggregates `gps` and `network` and potentially other sensors as well.
    // Unfortunately, certain LineageOS ROMs have broken `fused` provider that pretends to be enabled, but in
    // reality it does absolutely nothing and doesn't return any location updates. For this reason, we try all
    // (`fused`, `network`, `gps`) providers here, but prefer `fused` in LocationHelper.onLocationChanged().
    //
    // https://developer.android.com/reference/android/location/LocationManager#FUSED_PROVIDER
    // https://issuetracker.google.com/issues/215186921#comment3
    // https://github.com/organicmaps/organicmaps/issues/4158
    //
    mProviders.addAll(mLocationManager.getProviders(true));
    mProviders.remove(LocationManager.PASSIVE_PROVIDER); // not really useful if other providers are enabled.
    if (mProviders.isEmpty())
    {
      // Call this callback in the next event loop to allow LocationHelper::start() to finish.
      Logger.e(TAG, "No providers available");
      runLater(mListener::onLocationDisabled);
      return;
    }

    for (String provider : mProviders)
    {
      Logger.d(TAG, "Request Android native provider '" + provider
               + "' to get locations at this interval = " + interval + " ms");
      LocationManagerCompat.requestLocationUpdates(mLocationManager, provider, locationRequest,
          mNativeLocationListener, Looper.myLooper());
    }
  }

```

**Comment by rtsisyk** [comment]
> why our code doesn't check if network provider is enabled before subscribing to it

It does. `mLocationManager.getProviders(true)` returns only enabled providers.

https://developer.android.com/reference/android/location/LocationManager#getProviders(boolean)

> public [List](https://developer.android.com/reference/java/util/List)<[String](https://developer.android.com/reference/java/lang/String)> getProviders (boolean enabledOnly)
> enabledOnly | boolean: if true then only enabled providers are included

**Comment by rtsisyk** [comment]
> What about implementing a fallback similar to what has been done to fix this issue [mrmans0n/smart-location-lib#11](https://github.com/mrmans0n/smart-location-lib/issues/11) ?

Fallback to what? In the current implementation, we always subscribe to all enabled providers returned by the system.

> Here's what I think is the relevant bit. I can send you the full log by mail (I'm not willing to publish it, since it might contain private data).
> 
> ```
> 10-29 22:46:19.415  8472  8472 I app.organicmaps.location: 
> 10-29 22:46:19.418   601   601 I MotoSensors: Sensorhub hal setDelay: 19 - 66
> 10-29 22:46:19.418   601   601 I MotoSensors: Sensorhub hal setdelay: 19 - 5
> 10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateAccelRate 200
> 10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateGyroRate 200
> 10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateMagRate 200
> 10-29 22:46:19.419   601   601 I MotoSensors: Sensorhub hal enable: 19 - 1
> 10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateAccelRate 10
> 10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateGyroRate 5
> 10-29 22:46:19.419   601   601 I MotoSensors: HubSensors::updateMagRate 20
> 10-29 22:46:19.420  8472  8472 I app.organicmaps.location: provider = AndroidNativeProvider mInFirstRun = false oldInterval = 0 interval = 3000
> 10-29 22:46:19.426  8472  8472 I MwmApplication: Framework initialized
> 10-29 22:46:19.429   612   612 I LocSvc_GnssInterface: ===> setPositionMode_1_1 line 471 
> 10-29 22:46:19.429   612   660 E LocSvc_APIClientBase: onResponseCb:894] ERROR: 5 ID: 57
> 10-29 22:46:19.429   612   612 I LocSvc_GnssInterface: ===> start line 286 
> 10-29 22:46:19.429   612   612 I LocSvc_APIClientBase: locAPIStartTracking:342] start new session: 58
> 10-29 22:46:19.429   612   662 I LocSvc_api_v02: ---> locClientSendReq line 2325 QMI_LOC_REG_EVENTS_REQ_V02
> 10-29 22:46:19.435   612   662 I LocSvc_api_v02: ---> locClientSendReq line 2325 QMI_LOC_SET_OPERATION_MODE_REQ_V02
> 10-29 22:46:19.435   612  1599 I LocSvc_ApiV02: <--- globalRespCb line 213 QMI_LOC_SET_OPERATION_MODE_REQ_V02
> 10-29 22:46:19.436   612   662 I LocSvc_api_v02: ---> locClientSendReq line 2325 QMI_LOC_START_REQ_V02
> 10-29 22:46:19.436   601   816 D MotoSensors: gyroIntegration: quatMul bad
> 10-29 22:46:19.436  8472  8472 W onAccuracyChanged: Sensor android.sensor.rotation_vector has changed accuracy to 3
> 10-29 22:46:19.438   616   616 I android.hardware.power-service-qti: Power setMode: 5 to: 0
> 10-29 22:46:19.441   601   601 D MotoSensors: gyroIntegration: quatMul bad
> 10-29 22:46:19.442   612  1599 I LocSvc_ApiV02: <--- globalEventCb line 185 QMI_LOC_EVENT_WIFI_REQ_IND_V02
> 10-29 22:46:19.442   612  1599 I LocSvc_ApiV02: <--- globalEventCb line 185 QMI_LOC_EVENT_WIFI_REQ_IND_V02
> 10-29 22:46:19.442   612   660 I LocSvc_GnssInterface: ===> odcpiRequestCb line 513 
> 10-29 22:46:19.445   601   601 D MotoSensors: gyroIntegration: quatMul bad
> 10-29 22:46:19.447  2732  2749 I GnssLocationProvider: GNSS HAL Requesting location updates from network provider for 10000 millis.
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: Unable to request location.
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: java.lang.IllegalArgumentException: provider doesn't exist: network
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.location.LocationManagerService.requestLocationUpdatesLocked(LocationManagerService.java:1964)
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.location.LocationManagerService.requestLocationUpdates(LocationManagerService.java:1947)
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1178)
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1139)
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.location.gnss.GnssLocationProvider.handleRequestLocation(GnssLocationProvider.java:828)
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.location.gnss.GnssLocationProvider.access$2700(GnssLocationProvider.java:100)
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.location.gnss.GnssLocationProvider$ProviderHandler.handleMessage(GnssLocationProvider.java:2079)
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at android.os.Handler.dispatchMessage(Handler.java:106)
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at android.os.Looper.loop(Looper.java:223)
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at android.os.HandlerThread.run(HandlerThread.java:67)
> 10-29 22:46:19.450  2732  2749 W GnssLocationProvider: 	at com.android.server.ServiceThread.run(ServiceThread.java:44)
> 10-29 22:46:19.450   601   601 D MotoSensors: gyroIntegration: quatMul bad
> 10-29 22:46:19.504   601   601 I chatty  : uid=1000(system) sensors@1.0-ser identical 11 lines
> ```

Thanks for troubleshooting that. It looks like that the system ( [LocationManager.getProviders()](https://developer.android.com/reference/android/location/LocationManager#getProviders(boolean)) is the system Android API) returns `network` provider but then crashes on attempt to use to it. Probably the same story happens with `fused` that reported to be enabled, but never returns location. I think that we need to get LineageOS flashed to one of the devices to experiment with workarounds for LineageOS bugs.

**Comment by rtsisyk** [comment]
Wait, we never use 10 seconds interval:

> GNSS HAL Requesting location updates from network provider for 10000 millis.

Before subscribing to location, Organic Maps should print something like that:

>I app.organicmaps.location: LocationHelper.java:296 start(): provider = AndroidNativeProvider mInFirstRun = false oldInterval = 0 interval = 0
> D AndroidNativeProvider: AndroidNativeProvider.java:113 start(): Request Android native provider 'fused' to get locations at this interval = 0 ms
> D AndroidNativeProvider: AndroidNativeProvider.java:113 start(): Request Android native provider 'gps' to get locations at this interval = 0 ms
> D AndroidNativeProvider: AndroidNativeProvider.java:113 start(): Request Android native provider 'network' to get locations at this interval = 0 ms 

I don't see such lines in the log. I suspect that "W GnssLocationProvider: java.lang.IllegalArgumentException: provider doesn't exist: network" crash is caused by some other app working in parallel with Organic Maps. Otherwise you should see "Request Android native provider 'network' to get locations at this interval = 0 ms"  message before the crash.

**Comment by biodranik** [comment]
@rtsisyk OSMand works on Lineage (@jech right?), should we again check its source code to understand why?

**Comment by jech** [comment]
> Before subscribing to location, Organic Maps should print something like that:
> 
> I app.organicmaps.location: LocationHelper.java:296 start(): provider = AndroidNativeProvider mInFirstRun = false oldInterval = 0 interval = 0

There's no mention of LocationHelper in the full log.  Do I need to do something specific in OM?  Currently, I'm just running OM then quitting and finally running `adb logcat`.

**Comment by rtsisyk** [comment]
> > Before subscribing to location, Organic Maps should print something like that:
> > I app.organicmaps.location: LocationHelper.java:296 start(): provider = AndroidNativeProvider mInFirstRun = false oldInterval = 0 interval = 0
> 
> There's no mention of LocationHelper in the full log. Perhaps I need to do something in Organic Maps to see the `D` messages?
> 
> > is caused by some other app working in parallel with Organic Maps
> 
> It's reproducible, it happens every time I start Organic Maps. If you look at the log above, it happens 50ms after the line `app.organicmaps.location: provider = AndroidNativeProvider ...`, which IMHO is a strong indication (but not a proof) that it's triggerred by Organic Maps.
> 
> > OSMand works on Lineage (@jech right?)
> 
> Yes.

It looks like that debug logs are not printed to logcat because of #5943. Please try to enable Settings -> Enable logging to get the full picture here.

**Comment by jech** [comment]
> Please try to enable Settings -> Enable logging to get the full picture here.

Curiouser and curiouser.  It looks like Organic Maps correctly determined that the fused and network providers are not available:
```
Location providers: passive gps
```
A bit later, it said something unexpected:
```
11-01 00:25:11.164 (main) D/MwmApplication: onForeground(): 
11-01 00:25:11.167 (main) I/app.organicmaps.location: start(): 
11-01 00:25:11.169 (main) D/app.organicmaps.location: checkForAgpsUpdates(): A-GPS should be up to date
11-01 00:25:11.170 (main) D/SensorHelper: start(): 
11-01 00:25:11.173 (main) D/app.organicmaps.location: calcLocationUpdatesInterval(): mode = 0 interval = 3000
11-01 00:25:11.176 (main) I/app.organicmaps.location: start(): provider = AndroidNativeProvider mInFirstRun = false oldInterval = 0 interval = 3000
11-01 00:25:11.184 (main) D/AndroidNativeProvider: start(): 
11-01 00:25:11.185 (main) D/AndroidNativeProvider: start(): Request Android native provider 'gps' to get locations at this interval = 3000 ms
11-01 00:25:11.187 (main) I/MwmApplication: initNativeFramework(): Framework initialized
```
Two seconds later:
```
11-01 00:25:13.498 (main) I/app.organicmaps.location: start(): 
11-01 00:25:13.500 (main) D/app.organicmaps.location: checkForAgpsUpdates(): A-GPS should be up to date
11-01 00:25:13.503 (main) D/SensorHelper: start(): 
11-01 00:25:13.504 (main) D/app.organicmaps.location: calcLocationUpdatesInterval(): mode = 0 interval = 3000
11-01 00:25:13.506 (main) I/app.organicmaps.location: start(): provider = AndroidNativeProvider mInFirstRun = false oldInterval = 3000 interval = 3000
11-01 00:25:13.507 (main) D/AndroidNativeProvider: start(): 
11-01 00:25:13.510 (main) D/AndroidNativeProvider: start(): Request Android native provider 'gps' to get locations at this interval = 3000 ms
```
Then again:
```
11-01 00:25:26.890 (main) I/app.organicmaps.location: start(): 
11-01 00:25:26.891 (main) D/app.organicmaps.location: checkForAgpsUpdates(): A-GPS should be up to date
11-01 00:25:26.892 (main) D/SensorHelper: start(): 
11-01 00:25:26.895 (main) D/app.organicmaps.location: calcLocationUpdatesInterval(): mode = 0 interval = 3000
11-01 00:25:26.896 (main) I/app.organicmaps.location: start(): provider = AndroidNativeProvider mInFirstRun = false oldInterval = 0 interval = 3000
11-01 00:25:26.898 (main) D/AndroidNativeProvider: start(): 
11-01 00:25:26.899 (main) D/AndroidNativeProvider: start(): Request Android native provider 'gps' to get locations at this interval = 3000 ms
```

**Comment by biodranik** [comment]
Seeing the full log would be helpful.

**Comment by rtsisyk** [comment]
> Location providers: passive gps
> D/AndroidNativeProvider: start(): Request Android native provider 'gps' to get locations at this interval = 3000 ms

Yes, that confirms that Organic Maps subscribes ONLY to `gps` provider without `fused` and `network`. The issue with `java.lang.IllegalArgumentException: provider doesn't exist: network` is probably caused by some other app or a system service. 

gps takes some time to warm up. Organic Maps waits for gps only when it is on the screen or when navigation is enabled. The location indicator will not appear if the app is in the background without enabled navigation because the app disables location when goes to the background. This is not a bug. Please read https://github.com/organicmaps/organicmaps/issues/4223#issuecomment-1784225506.

**Comment by biodranik** [comment]
As far as I understood, the "GPS active" location indicator does not appear when OM is in the foreground, @jech right? I also observed the same behavior on some of my devices several times (unfortunately, don't remember the details).

**Comment by jech** [comment]
> As far as I understood, the "GPS active" location indicator does not appear when OM is in the foreground, @jech right?

That's correct: Organic Maps in the foreground, location enabled, the blue spinner spinning, but the location indicator does not appear.  Here's a screenshot:

![om](https://github.com/organicmaps/organicmaps/assets/290341/42d93df3-c286-472c-913c-da22eb850c0b)

**Comment by rtsisyk** [comment]
> > As far as I understood, the "GPS active" location indicator does not appear when OM is in the foreground, @jech right?
> 
> That's correct: Organic Maps in the foreground, location enabled, the blue spinner spinning, but the location indicator does not appear. Here's a screenshot:
> 
> ![om](https://user-images.githubusercontent.com/290341/279812321-42d93df3-c286-472c-913c-da22eb850c0b.png)

I am investigating this issue.

**Comment by rtsisyk** [comment]
I can confirm that with Organic Maps the green dot location indicator is shown only AFTER the location has been initially found. The app searches for the location, but the indicator is not displayed. I will try to dig into Android sources to figure out why it works in this way.

**Comment by rtsisyk** [comment]
This issue doesn't happen on Samsung with Android 12, but happens on CalyxOS with Android 13. It can be a regression caused by CalyxOS / LineageOS patches.

**Comment by rtsisyk** [comment]
LineageOS 18.1 is Android 11, so it is definitely not an Android 13 regression. I will try flash LineageOS instead of CalyxOS on my device to test.

**Comment by rtsisyk** [comment]
OSMAnd always has this indicator because they also subscribe to low-level GNSS status updates (i.e. the number of satelleites, signal, etc.) from the system. It doesn't

**Comment by jech** [comment]
So what you're saying is that even with the current release, the GPS will eventually get a fix even though the location indicator doesn't appear?  Interesting, it never occurred to me that could be the case.

I'll test this theory tomorrow.

**Comment by biodranik** [comment]
Can you please try outdoors?

**Comment by jech** [comment]
It looks like you're right. I first used gpstest to get a fix, which took 
34s. I then killed gpstest, waited for ten seconds or so, then launched 
OM, which got a fix after a few seconds.

So it looks like the problem is only the indicator not activating before 
we get a fix. Which I think is a fairly serious bug on LineageOS, but 
that's a different matter.

Thanks again for your help.
